frontend/src/components/Wallet.tsx

```diff
@@ -0,0 +1,51 @@
+import { CheckIcon, CopyIcon } from "@chakra-ui/icons";
+import { Box, Center, Divider, HStack, IconButton, Skeleton, Text, useClipboard } from "@chakra-ui/react";
+import React from "react";
+import { unixTimestampToDate, WalletInfo } from "./Types";
+
+interface WalletProps {
+    walletInfo: WalletInfo | null;
+}
+
+export default function Wallet(
+    {
+        walletInfo,
+    }: WalletProps,
+) {
+    const { hasCopied, onCopy } = useClipboard(walletInfo ? walletInfo.address : "");
+
+    let balance = <Skeleton height="20px" />;
```

Comment:
I'd prefer conditional definitions over mutable variables:

```suggestion
    const balance = walletInfo ? <Text>{walletInfo.balance} BTC</Text> : <Skeleton height="20px" />;
```

Reply:
In this case I disagree. I think the code reads way nicer the way it is. You see that all the different widgets are initialized as `Skeleton` and only if `walletInfo` is present they change. Additionally less code duplication this way, because otherwise I have to check `walletInfo` 3 times...